src/EditorFeatures/Core/Implementation/Suggestions/SuggestedAction.cs

```diff
@@ -86,7 +90,39 @@ protected Task<ImmutableArray<CodeActionOperation>> GetPreviewOperationsAsync(Ca
                 async () => await CodeAction.GetPreviewOperationsAsync(cancellationToken).ConfigureAwait(false), cancellationToken);
         }
 
-        public virtual void Invoke(CancellationToken cancellationToken)
+        public void Invoke(CancellationToken cancellationToken)
```

Comment:
1. This can't be async.  It's the implementation of ISuggestedAction.Invoke.
2. If this waits on the task, then it defeats the purpose.  We're back to blocking with the lightbulb up while the invocatoin happens.

This is actually a case of 'fire and forget'.  The editor fires and forgets (and dismisses the light bulb).  We then make a TWD and we do our work behind it.  

Your point about exceptions is valid though.  I'm happy to add a continuation that reports the exception properly.  Is that sufficient for you?